signer/tests/integration/contracts.rs

```diff
@@ -114,6 +120,36 @@ impl SignerStxState {
             SubmitTxResponse::Rejection(err) => panic!("{}", serde_json::to_string(&err).unwrap()),
         }
     }
+
+    /// Get the source of the a deployed smart contract.
+    ///
+    /// # Notes
+    ///
+    /// This is useful just to know whether a contract has been deployed
+    /// already or not. If the smart contract has not been deployed yet,
+    /// the stacks node returns a 404 Not Found.
+    async fn get_contract_source<T>(&self) -> Result<ContractSrcResponse, reqwest::Error>
```

Comment:
Tbh it wouldn't hurt to have this in the actual `StacksClient`, you're already half-using it

Reply:
Honestly, I got lazy and didn't want to add the tests that I normally would for a proper function for the `StacksClient` 😅. Plus I wanted to keep this PR focused. But this is a good point, I'll add it in another PR though.

Reply:
Haha okay, was just a suggestion anyway, i'll resolve this comment then :)